scripts/maintenance/standard-header-guards.pl

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/perl
+
+## Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+# USAGE: standard-header-new_guards.pl filename.h >filename.h.new
+#
+# source-maintenance plungin to standardise the header guards
+#
+# Behaviour:
+# - files whose name does not end in .h are passed verbatim
+# - anything before the first #ifdef SOMETHING_ENDING_IN_H is copied verbatim
+# - the existing header guard is replaced with SQUID_RELATIVE_PATH_OF_FILE_H
+# - anything until the last #endif clause in the file is copied verbatim
+# - the last #endif clause is replaced with the closing header guard and an empty line
+
+use strict;
+use warnings;
+
+my $filename = $ARGV[0];
+my $new_guard = $filename;
+$new_guard =~ s/[\/\-\.]/_/g;
+$new_guard = "SQUID_".uc($new_guard);
+
+unless ($filename =~ /\.h$/i) {
+    while (<>) { print; }
+    exit(0);
+}
+
+# print preamble
+while (<>) {
+    last if (/^#ifndef\s+(\w+_H)/);
+    print;
+}
+
+# we are at end of fine, no preamble found
+exit(0) unless defined($_);
```

Comment:
Please quit with an error if a `.h` file in `src/` does not have a guard (that the script can detect). I counted approximately 15 such files (that all should be fixed AFAICT):

```
$ git grep -L -P '^#ifndef\s+(\w+_H)' '*.h' | fgrep src/
src/acl/AnnotateClient.h
src/acl/AnnotateTransaction.h
src/acl/external/kerberos_ldap_group/support.h
src/adaptation/ecap/Registry.h
src/auth/State.h
src/auth/basic/NIS/nis_support.h
src/auth/basic/RADIUS/radius-util.h
src/auth/basic/RADIUS/radius.h
src/auth/digest/LDAP/ldap_backend.h
src/auth/digest/eDirectory/edir_ldapext.h
src/auth/digest/eDirectory/ldap_backend.h
src/auth/digest/file/text_backend.h
src/auth/negotiate/kerberos/negotiate_kerberos.h
src/tests/STUB.h
src/tests/TestSwapDir.h
```

P.S. Please fix the "fine" typo in the comment while you are adjusting this code.